packages/core/src/application.ts

```diff
@@ -83,7 +87,8 @@ export class Application extends Context implements LifeCycleObserver {
   constructor(configOrParent?: ApplicationConfig | Context, parent?: Context) {
     super(
       configOrParent instanceof Context ? configOrParent : parent,
-      'application',
+      // Default the context name to application name
+      configOrParent instanceof Context ? undefined : configOrParent?.name,
```

Comment:
What if the application name is not provided? IIUC, before your change, we would use `application` as the context name. With your change in place, we will use an auto-generated UUID instead.

Did you perhaps mean this?

```suggestion
      configOrParent instanceof Context ? 'application' : configOrParent?.name,
```

Also what if `name` is not set? I think we should use `application` in that case too.

TBH, I find these repeated `instanceof` checks a bit difficult to comprehend, let's normalize the arguments first before we start processing them.

```ts
export class Application extends Context implements LifeCycleObserver {
  //...

  constructor(configOrParent?: ApplicationConfig | Context, parent?: Context) {
    let config: ApplicationConfig;
    if (configOrParent instanceof Context) {
      parent = configOrParent;
      config = undefined;
    }

    config = config ?? {};

    super(parent, config.name ?? 'application');
    this.options = config;
  
    //...
  }

  //...
}
```

Reply:
Without a name, it will become `Application-<uuid>`.